### PR TITLE
Don't merge in Hound defaults

### PR DESCRIPTION
**Pull request by salbertson**
This change should make configuration easier. Our approach has changed from opinionated defaults to simply relying on linter defaults.

### CODE REVIEW

File: spec/models/style_guide/coffee_script_spec.rb [hunk 1]
```diff
@@ -45,125 +15,53 @@
...
-    context "with thoughtbot configuration" do
-      context "for an empty function" do
-        it "returns a file review without violations" do
-          code = <<-CODE.strip_heredoc
-            class FooBar
-              foo: ->
-          CODE
+    context "with custom configuration" do
+      context "when line length is configured" do
+        it "does not find line lenght violation" do
```
Comment (thorncp) [inline spec/models/style_guide/coffee_script_spec.rb]: `lenght` -> `length`

**Comment by thorncp** [comment]
Looks good! There are mentions of the default configs on the configuration page that should be removed as well.

File: spec/models/style_guide/coffee_script_spec.rb (hunk 1 above)
Comment (salbertson) [inline spec/models/style_guide/coffee_script_spec.rb]: :eyes:

File: app/models/style_guide/java_script.rb [hunk 2]
```diff
@@ -23,27 +21,19 @@ def file_included?(commit_file)
     private
 
     def config
-      custom_config = repo_config.for(name)
       if custom_config["predef"].present?
         custom_config["predef"] |= default_config["predef"]
       end
```
Comment (gylaz) [inline app/models/style_guide/java_script.rb]: Any reason we are keeping this? Aren't we removing `default_config`?

File: spec/models/repo_config_spec.rb [hunk 3]
```diff
@@ -175,6 +175,30 @@
...
+
+      context "when legacy coffee_script key is used to disable" do
+        it "returns false" do
+          commit = double("Commit", file_content: <<-EOS.strip_heredoc)
+            coffee_script:
+              enabled: false
+          EOS
+          repo_config = RepoConfig.new(commit)
+
+          expect(repo_config.enabled_for?("coffeescript")).to eq false
```
Comment (gylaz) [inline spec/models/repo_config_spec.rb]: What do you think about?
```rb
expect(repo_config).not_to be_enabled_for("coffeescript")
```

File: spec/models/style_guide/coffee_script_spec.rb [hunk 4]
```diff
@@ -45,125 +15,55 @@
...
 
-          expect(violations.size).to eq 1
-          expect(violation.filename).to eq "test.coffee"
-          expect(violation.patch_position).to eq 2
+          violation = violations.first
           expect(violation.line_number).to eq 1
-          expect(violation.messages).to match_array(
-            ["Line exceeds maximum allowed length"]
+          expect(violation.messages).to(
+            include("Line exceeds maximum allowed length")
```
Comment (gylaz) [inline spec/models/style_guide/coffee_script_spec.rb]: How about writing this as:
```rb
expect(violation.messages).to include(
  "Line exceeds maximum allowed length",
)
```
to follow some of the other specs?

File: spec/models/style_guide/coffee_script_spec.rb [hunk 5]
```diff
@@ -45,125 +15,53 @@
...
 
-        expect(violations.size).to eq 1
-        expect(violation.filename).to eq "test.coffee.erb"
-        expect(violation.messages).to match_array [
-          "Class name should be UpperCamelCased",
-        ]
+        violation = violations.first
+        expect(violation.line_number).to eq 1
+        expect(violation.messages).to(
+          include("Class name should be UpperCamelCased")
```
Comment (gylaz) [inline spec/models/style_guide/coffee_script_spec.rb]: Same suggestion here.

File: spec/models/style_guide/haml_spec.rb [hunk 6]
```diff
@@ -18,8 +18,47 @@
...
+        it "finds violations" do
+          content = "%div#container Hello"
+          config = {
+            "linters" => {
+              "ImplicitDiv" => {
+                "enabled" => true
+              }
+            }
+          }
+
```
Comment (gylaz) [inline spec/models/style_guide/haml_spec.rb]: Is this an unnecessary newline?

**Comment by gylaz** [comment]
Looks good to me. What is the plan for releasing this? Let it sit until we wire up a migration strategy?

**Comment by salbertson** [comment]
> What is the plan for releasing this? Let it sit until we wire up a migration strategy?

@gylaz yea, plan is to not merge it until we have a migration path in place for existing users to maintain their configuration. I'll keep the branch up to date with master until then.

**Pull request closed without merging**